crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -268,20 +268,21 @@ impl Settlement {
         auction: &competition::Auction,
         revert_protection: &mempools::RevertProtection,
     ) -> Result<competition::Score, score::Error> {
-        let objective_value = self
-            .boundary
-            .objective_value(eth, auction, self.gas.estimate)?;
+        let quality = self.boundary.quality(eth, auction)?;
 
         let score = match self.boundary.score() {
-            competition::SolverScore::Solver(score) => competition::Score(score),
+            competition::SolverScore::Solver(score) => competition::Score(score.try_into()?),
```

Comment:
I think one of these `.try_into()?` lead to a `/notify` request not being sent.
For example in this [run loop](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2023-10-31T19:37:22.545Z',to:'2023-10-31T19:49:56.704Z'))&_a=(columns:!(log),filters:!(),grid:(columns:('@timestamp':(width:164))),index:ea511870-d9b3-11ed-a9d0-a17451f01cc1,interval:auto,query:(language:kuery,query:'log:%20%2211712235%22%20and%20log:%20%22seasolver%22'),sort:!(!('@timestamp',desc))))

Reply:
Let me check...

Reply:
fixed error handling https://github.com/cowprotocol/services/pull/2031

How we ended up with [score::Solver(0)](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/doc/ea511870-d9b3-11ed-a9d0-a17451f01cc1/cowlogs-staging-2023.10.31?id=rLRBh4sBQgsUfyaRQinM) is still an enigma for me.

Reply:
Apparently seasolver used `scoreDiscount` variant which is deprecated and the deserialization should fallback to `RiskAdjusted` variant with `1.0` success probability.